tests/test-coauthors-plus.php

```diff
@@ -679,6 +699,1434 @@ public function test_update_author_term_when_author_term_not_exist() {
+	 * appending authors to the post, we should
+	 * expect the post_author to change.
+	 *
+	 * @return void
+	 */
+	public function test_assign_post_authors_from_authors_who_have_not_been_linked() {
+		$post_id = $this->factory()->post->create(
+			array(
+				'post_author' => $this->author1->ID,
+				'post_status' => 'publish',
+				'post_type'   => 'post',
+			)
+		);
+
+		$this->_cap->add_coauthors(
+			$post_id,
+			array(
+				$this->author3->user_login,
+				$this->editor1->user_login,
+				$this->author2->user_login,
+			)
+		);
+
+		$query = new WP_Query(
+			array(
+				'p' => $post_id,
+			)
+		);
+
+		$this->assertEquals( 1, $query->found_posts );
+		$this->assertEquals( $this->author3->ID, $query->posts[0]->post_author );
+
+		$post_author_terms = wp_get_post_terms( $post_id, $this->_cap->coauthor_taxonomy );
+
+		$this->assertTrue( is_array( $post_author_terms ) );
+		$this->assertCount( 3, $post_author_terms );
+
+		$author_slugs = array(
+			'cap-' . $this->author3->user_login,
+			'cap-' . $this->editor1->user_login,
+			'cap-' . $this->author2->user_login,
+		);
+
+		foreach ( $post_author_terms as $term ) {
+			$this->assertInstanceOf( WP_Term::class, $term );
+			$this->assertContains( $term->slug, $author_slugs );
+		}
+	}
+
+	/**
+	 * Here we are creating guest authors (coauthors) and assigning them to a post,
+	 * which was created by a WP_User. Since the guest authors have not been
+	 * linked to a WP_User, the wp_post.post_author column should not
+	 * change, and the response from CoAuthors_Plus::add_coauthors()
+	 * should be false, since no WP_User could be found.
+	 *
+	 * @return void
+	 */
+	public function test_assign_post_authors_from_coauthors_who_have_not_been_linked() {
+		$random_username = 'random_user_' . rand( 1, 1000 );
+		$display_name = str_replace( '_', ' ', $random_username );
+
+		$guest_author_1_id = $this->_cap->guest_authors->create(
+			array(
+				'user_login'   => $random_username,
+				'display_name' => $display_name,
+			)
+		);
+		/**
+		 * By using CoAuthors_Plus::get_coauthor_by(), we are ensuring
+		 * that the recent changes to the code will prioritize
+		 * returning a Guest Author when one is found.
+		 */
+		$guest_author_1 = $this->_cap->get_coauthor_by( 'id', $guest_author_1_id );
+
+		$this->assertIsObject( $guest_author_1 );
+		$this->assertThat(
+			$guest_author_1,
+			$this->logicalNot(
+				$this->isInstanceOf( WP_User::class )
+			)
+		);
+
+		$random_username = 'random_user_' . rand( 1001, 2000 );
+		$display_name = str_replace( '_', ' ', $random_username );
+
+		$guest_author_2_id = $this->_cap->guest_authors->create(
+			array(
+				'user_login'   => $random_username,
+				'display_name' => $display_name,
+			)
+		);
+		$guest_author_2 = $this->_cap->get_coauthor_by( 'id', $guest_author_2_id );
+
+		$this->assertIsObject( $guest_author_2 );
+		$this->assertThat(
+			$guest_author_2,
+			$this->logicalNot(
+				$this->isInstanceOf( WP_User::class )
+			)
+		);
```

Comment:
Shouldn't need these assertions again - we already run these assertions earlier in this test method, and there's no reason to suspect that the behaviours of `get_coauthor_by()` will have changed.